### PR TITLE
Don’t coerce a date type set with `null` to be 1970.

### LINKED ISSUE #128: Don't coerce null in dataType `date` as 0
Getting `key: null` from server for data type `date` coerces into `1970.....`

### PR DESCRIPTION
**Pull request by latentflip**
fixes #128 

Since we try and coerce in both get and set I think we need this in both places.

### CODE REVIEW

**Comment by HenrikJoreteg** [comment]
Should we use  `allowNull`?

**Comment by latentflip** [comment]
@HenrikJoreteg I don't think allowNull is relevant here. The coercion happens before the null check. So even as it was, if you set required and allowNull to true, it would still coerce. And even if allowNull is true, you don't want it coerced to 1970.

**Comment by latentflip** [comment]
Added a test to verify that allowNull:false still works as expected on dates.

File: ampersand-state.js [hunk 1]
```diff
@@ -604,13 +606,18 @@ var dataTypes = {
...
+
             return {
                 val: newVal,
                 type: newType
             };
         },
         get: function (val) {
-            return new Date(val);
+            if (val == null) {
+                return undefined;
```
Comment (HenrikJoreteg) [inline ampersand-state.js]: I wonder if this should just return `null` instead?

**Comment by HenrikJoreteg** [comment]
this makes sense to me +1

**Comment by bear** [comment]
does this require a documentation PR to note the change in behaviour?

**Comment by HenrikJoreteg** [comment]
To me, this feels like a bug we're fixing that should've already been how it behaved. But, I would agree that it's probably worth clarifying how this case is handled in the docs.

**Comment by kamilogorek** [comment]
Looks fine for me +1

**Pull request merged**